### PR TITLE
[Wallet]: Add error message and retry for swap errors

### PR DESCRIPTION
**Pull request by plam-ml**
## Description 

https://mysten.atlassian.net/browse/APPS-2049
error-swap

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/shared/InputWithAction.tsx [hunk 1]
```diff
@@ -214,12 +221,30 @@ export const InputWithActionButton = forwardRef<HTMLInputElement, InputWithActio
...
+							loading && 'text-transparent',
 						)}
 						disabled={disabled}
 						ref={forwardRef}
 					/>
+					{loading && (
+						<div className="absolute">
+							<div className="flex items-center gap-1 text-steel">
+								<LoadingIndicator color="inherit" />
+								{loadingText && (
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/InputWithAction.tsx:233]: You don't use this below, it's hardcoded to `Calculating`. Also from what I can tell this is always in a readonly state so it's kind of silly that it's even an input to begin with (don't need to change this just think it's interesting.

File: apps/wallet/src/ui/app/shared/InputWithAction.tsx [hunk 2]
```diff
@@ -170,9 +171,12 @@ type InputWithActionZodFormProps = VariantProps<typeof inputWithActionZodFormSty
 		errorString?: string;
 		suffix?: ReactNode;
 		prefix?: ReactNode;
+		loading?: boolean;
+		loadingText?: string;
 		onActionClicked?: PillProps['onClick'];
 		info?: ReactNode;
 		actionDisabled?: boolean;
+		placeholder?: string;
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/InputWithAction.tsx:179]: This prop looks unused

**Review by Jordan-Mysten — COMMENTED** [review]
Two thoughts:
- What's the case where this errors out (even with the retries we have baked in). but clicking "retry" succeed?
- It's weird to me to have a button to retry, can we not just keep retrying with some larger timeout?

**Comment by plam-ml** [comment]
> Two thoughts:
> 
> * What's the case where this errors out (even with the retries we have baked in). but clicking "retry" succeed?
> * It's weird to me to have a button to retry, can we not just keep retrying with some larger timeout?

@Jordan-Mysten We can use a large number of retries. But I think the main thing is to surface the error message so that we can see what's wrong in the case someone submits a screenshot. If we think the button is pointless, I can chat with Morgan and see if we can simply remove it.

File: apps/wallet/src/ui/app/shared/InputWithAction.tsx (hunk 1 above)
Comment (plam-ml) [reply apps/wallet/src/ui/app/shared/InputWithAction.tsx:231]: updated

**Review by Jordan-Mysten — APPROVED** [review]

**Pull request merged**